src/actladder.cpp

```diff
@@ -786,56 +857,81 @@ void Entity::actMidGamePortal()
-					}
-				}
-				int introstageToChangeTo = 9;
-				if ( movieCrawlType >= 0 )
-				{
-					introstageToChangeTo = 11 + movieCrawlType;
+				    cutscene = 1;
 				}
 
 				if ( multiplayer == SERVER )
 				{
-					for ( c = 1; c < MAXPLAYERS; c++ )
+					for ( int c = 1; c < MAXPLAYERS; c++ )
 					{
 						if ( client_disconnected[c] == true || players[c]->isLocalPlayer() )
 						{
 							continue;
 						}
 						strcpy((char*)net_packet->data, "MIDG");
-						net_packet->data[4] = introstageToChangeTo;
+						net_packet->data[4] = cutscene;
 						net_packet->address.host = net_clients[c - 1].host;
 						net_packet->address.port = net_clients[c - 1].port;
-						net_packet->len = 8;
+						net_packet->len = 5;
 						sendPacketSafe(net_sock, -1, net_packet, c - 1);
 					}
 				}
-				subwindow = 0;
-				fadeout = true;
-				if ( !intro )
-				{
-					pauseGame(2, false);
-				}
-				introstage = introstageToChangeTo; // prepares mid game sequence
+
+	            if (cutscene == 0) {
+	                switch (stats[clientnum]->playerRace) { // herx midpoint
+	                default:
+	                case RACE_HUMAN:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::HerxMidpointHuman);
+	                    break;
+	                case RACE_AUTOMATON:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::HerxMidpointAutomaton);
+	                    break;
+	                case RACE_GOATMAN:
+	                case RACE_GOBLIN:
+	                case RACE_INSECTOID:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::HerxMidpointBeast);
+	                    break;
+	                case RACE_SKELETON:
+	                case RACE_VAMPIRE:
+	                case RACE_SUCCUBUS:
+	                case RACE_INCUBUS:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::HerxMidpointEvil);
+	                    break;
+	                }
+	            }
+	            else if (cutscene == 1) { // baphomet midpoint
+	                switch (stats[clientnum]->playerRace) {
+	                default:
+	                case RACE_HUMAN:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::BaphometMidpointHuman);
+	                    break;
+	                case RACE_AUTOMATON:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::BaphometMidpointAutomaton);
+	                    break;
+	                case RACE_GOATMAN:
+	                case RACE_GOBLIN:
+	                case RACE_INSECTOID:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::BaphometMidpointBeast);
+	                    break;
+	                case RACE_SKELETON:
+	                case RACE_VAMPIRE:
+	                case RACE_SUCCUBUS:
+	                case RACE_INCUBUS:
+	                    MainMenu::beginFade(MainMenu::FadeDestination::BaphometMidpointEvil);
+	                    break;
+	                }
+	            }
+
+                movie = true;
```

Comment:
Reminder to check Entity::isMobile to make sure monsters don't move around in cutscenes